iceoryx_hoofs/source/posix_wrapper/mutex.cpp

```diff
@@ -74,7 +74,8 @@ bool mutex::unlock()
 
 bool mutex::try_lock()
 {
-    return !posixCall(pthread_mutex_trylock)(&m_handle).successReturnValue(0).evaluate().has_error();
+    auto result = posixCall(pthread_mutex_trylock)(&m_handle).successReturnValue(0).evaluateWithIgnoredErrnos(EBUSY);
+    return !(!result.has_error() && result->errnum == EBUSY) && !result.has_error();
```

Comment:
The `!(!result.has_error() && result->errnum == EBUSY)` part is hard to graps. Would you mind introducing a variable like
```cpp
auto isEBUSY = !result.has_error() && result->errnum == EBUSY;
return !isEBUSY && !result.has_error();
```